pkg/config/mock/mock.go

```diff
@@ -70,3 +71,33 @@ func New(t testing.TB) model.Config {
 	setup.InitConfig(newCfg)
 	return &mockConfig{newCfg}
 }
+
+// NewSystemProbe creates a mock for the system-probe config
+func NewSystemProbe(t testing.TB) model.Config {
+	// We only check isSystemProbeConfigMocked when registering a cleanup function. 'isSystemProbeConfigMocked'
+	// avoids nested calls to Mock to reset the config to a blank state. This way we have only one mock per test and
+	// test helpers can call Mock.
+	if t != nil {
+		m.Lock()
+		defer m.Unlock()
+		if isSystemProbeConfigMocked {
+			// The configuration is already mocked.
+			return &mockConfig{setup.SystemProbe()}
+		}
+
+		isSystemProbeConfigMocked = true
+		originalConfig := setup.SystemProbe()
+		t.Cleanup(func() {
+			m.Lock()
+			defer m.Unlock()
+			isSystemProbeConfigMocked = false
```

Comment:
Can you explain why `isSystemProbeConfigMocked` is protected by a lock and not all calls to `setup.SystemProbe()` and `setup.SetSystemProbe` which also use a global state.